agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
+         */
+        Type instanceType = invocation.getReferenceType(pgen);
+        Type[] arguments = BCELUtil.insert_type(instanceType, invocation.getArgumentTypes(pgen));
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                arguments,
+                Const.INVOKESTATIC);
+        break;
+      case Const.INVOKESTATIC:
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC);
+        break;
       default:
-        return (null);
+        debug_transform.log(
+            "getReplacementInstruction: EXIT wrong instruction type in %s.%s%n",
+            mg.getClassName(), mg.getName());
+        return null;
     }
+    debug_transform.log("getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+    return instruction;
   }
 
   /**
-   * Replace instruction ih in list il with the instructions in new_il. If new_il is null, do
-   * nothing.
+   * Returns the replacement method for the given method if one is determined by a method, class or
+   * package replacement, or null otherwise.
    *
-   * @param il the instruction list
-   * @param ih the instruction
-   * @param new_il the new instructions to substitute
+   * @param methodDef the method to replace, must not be null
+   * @return the replacement method, null if there is none
    */
-  protected static void replace_instructions(
-      InstructionList il, InstructionHandle ih, InstructionList new_il) {
+  private MethodDef getReplacementMethod(MethodDef methodDef) {
+    return replacementMap.get(methodDef);
+  }
 
-    if ((new_il == null) || new_il.isEmpty()) {
-      return;
+  /**
+   * Replace the instruction in the original instruction list with the instructions in the new
+   * instruction list. If this instruction was the target of any jumps or line numbers, replace them
+   * with the first instruction in the new list.
+   *
+   * @param instructionList the original instruction list, must be non-null
+   * @param instruction the instruction to replace
+   * @param invokeInstruction the new instructions to substitute for the instruction, must be
+   *     non-null and non-empty
+   */
+  private static void replaceInstructions(
+      InstructionList instructionList,
+      InstructionHandle instruction,
+      InvokeInstruction invokeInstruction) {
+
+    if ((invokeInstruction == null)) {
+      throw new IllegalArgumentException("New instruction must be non-null");
     }
 
     // If there is only one new instruction, just replace it in the handle
-    if (new_il.getLength() == 1) {
-      ih.setInstruction(new_il.getEnd().getInstruction());
+    if (invokeInstruction.getLength() == 1) {
```

Comment:
`getLength()` returns the number of bytes in the instruction.
On an instruction list, it returns the number of elements, but you changed the code to accept a single instruction rather than a list, without changing this code.
I don't think you need to handle the case of multiple instructions in this method any longer.